src/protagonist/Engine/Ingest/Image/ImageServer/Clients/CantaloupeThumbsClient.cs

```diff
@@ -84,6 +86,18 @@ public class CantaloupeThumbsClient : IThumbsClient
         throw new HttpException(response.StatusCode, "failed to retrieve data from the thumbs processor");
     }
 
+    private void AttemptToAddStickinessCookie(HttpResponseMessage response)
+    {
+        if (!cantaloupeClient.DefaultRequestHeaders.Contains("Cookie"))
+        {
+            var hasCookie = response.Headers.TryGetValues("Set-Cookie", out var cookies);
+            if (hasCookie)
+            {
+                cantaloupeClient.DefaultRequestHeaders.Add("Cookie", cookies!);
```

Comment:
Is there any danger of this being reused if it's on the `DefaultRequestHeaders`? IIRC some element of the HttpClient can be pooled and reused when using `HttpClientFactory` does `DefaultRequestHeaders` fit the bill? 

Would it to be better to set it per request, would mean we could handle the following from docs:
> If a target fails or becomes unhealthy, the load balancer stops routing requests to that target, and chooses a new healthy target based on the chosen load balancing algorithm. The load balancer treats the session as now being "stuck" to the new healthy target, and continues routing requests to the new healthy target even if the failed target comes back.

It's unlikely that could happen but if we send cookies returned from request A with request B, then cookies from B -> C etc (rather than using cookies returned from request A with B, C, D....K) then we'd always be using the latest.

Reply:
spent some time taking a look at this, and like you said unlikely but possible - think I went this way as it's using `GetAsync` - will change it to use `SendAsync`